webapp/src/Controller/MetricsController.php

```diff
@@ -0,0 +1,157 @@
+<?php declare(strict_types=1);
+
+namespace App\Controller;
+
+use App\Service\DOMJudgeService;
+use App\Service\SubmissionService;
+use App\Entity\Team;
+use Doctrine\ORM\EntityManagerInterface;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\IsGranted;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+use Prometheus\CollectorRegistry;
+use Prometheus\RenderTextFormat;
+
+/**
+ * @Route("/prometheus/metrics")
+ * @IsGranted("ROLE_ADMIN")
```

Comment:
Do we want to use a separate role for this?

Reply:
Or if not a separate role, api_reader maybe more appropriate than admin?

Reply:
I would say admin (or maybe api_reader) is correct: we're exposing general information about the system and I'd prefer to not have to worry about security implications here. Also, this would only be used by admins anyways.

Reply:
I would not want to use the full access admin account on another machine that only needs read access.

Its an API that only needs to be read from. And I believe it by and large exposes what is in another form available from the existing API. So I think the API readonly user is a good match.

Reply:
What is the scope of access for `api_reader`? Does it generally have access to "non-public" data/information during freezetime?

I made this admin just to make sure information doesn't leak if we added some metric like problems solved per team; but I do agree that putting the full admin account credentials on another machine is less than ideal.

I'm not opposed to creating a new metrics role or something like that and using it for just this one endpoint if we think that'd be the best solution.

Reply:
Api reader is what kattis/cds uses so it can see everything

Reply:
ok, that seems more reasonable to use then. I'll make that change a bit later(and fix up composer.lock)